### PR TITLE
[bug fix] hmr doesn't crash app if it fails

### PR DESCRIPTION
**Pull request by niicojs**
If hmr fails (in my case for example I was testing through ngrok), the whole site doesn't break.
I've just catch error and print it to console so the app can keep loading

### CODE REVIEW

**Comment by DeMoorJasper** [comment]
@davidnagli the hmr-runtime is a built-in (gets packaged in the bundle) and has no access to the Logger class?

**Comment by niicojs** [comment]
also, this is client side, `Logger` is while generating right?

**Comment by davidnagli** [comment]
Oh ya sorry didn’t realize this was the HMR runtime 😂

File: src/builtins/hmr-runtime.js [hunk 1]
```diff
@@ -15,38 +15,43 @@ function Module() {
...
 
-    if (data.type === 'error') {
-      console.error('[parcel] 🚨  ' + data.error.message + '\n' + 'data.error.stack');
-    }
-  };
+      if (data.type === 'error') {
+        console.error('[parcel] 🚨  ' + data.error.message + '\n' + data.error.stack);
+      }
+    };
+  } catch (e) {
```
Comment (devongovett) [inline src/builtins/hmr-runtime.js]: Do we need the try/catch around the entirety of this code or is only a certain line failing? Also what is the issue you had? Connection failure or something? What about a `ws.onerror` handler?

Comment (niicojs) [reply src/builtins/hmr-runtime.js]: I was using ngrok through https, so Chrome did not want to allow connection to non-secure websocket used by hmr. and it crashed my whole app.
for me, hmr can't crash the underlying app, that's why I made a (too?) broad try / catch.

Comment (DeMoorJasper) [reply src/builtins/hmr-runtime.js]: Isn't this fixed by the recent https enabled websocket?

**Comment by devongovett** [comment]
Due to move to monorepo which rewrote the git history, this PR was automatically closed. Please open a new PR against master with your changes. You can do a diff against the `1.x` branch to see them. Sorry about that!

**Pull request closed without merging**